src/wp-includes/class-wp-query.php

```diff
@@ -2544,18 +2545,19 @@ public function get_posts() {
 			}
 		} elseif ( ! empty( $post_type ) && is_array( $post_type ) ) {
 			$post_type_where = " AND {$wpdb->posts}.post_type IN ('" . implode( "', '", esc_sql( $post_type ) ) . "')";
-		} elseif ( ! empty( $post_type ) ) {
+		} else {
+			if ( empty( $post_type ) ) {
+				$post_type_prime = 'any';
+				if ( $this->is_attachment ) {
+					$post_type = 'attachment';
```

Comment:
Are we sure that altering the value of the `$post_type` variable here is safe? Seems to be used elsewhere in this function. For example, it looks like it could cause side effects for the capabilities checks @peterwilsoncc added in [51276](https://core.trac.wordpress.org/changeset/51276).

Reply:
In my testing, it wasn't a issue. If anything it would be a posative, as post type being empty doesn't really help ina lot of code paths.